jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/TriggerAsyncNodeTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.test.functional.async;
+
+import java.util.HashMap;
+
+import org.jbpm.executor.ExecutorServiceFactory;
+import org.jbpm.kie.services.impl.admin.commands.TriggerNodeCommand;
+import org.jbpm.test.JbpmTestCase;
+import org.jbpm.test.wih.FirstErrorWorkItemHandler;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.event.process.DefaultProcessEventListener;
+import org.kie.api.event.process.ProcessCompletedEvent;
+import org.kie.api.executor.ExecutorService;
+import org.kie.api.runtime.KieSession;
+
+
+public class TriggerAsyncNodeTest extends JbpmTestCase {
+
+
+    private static final String PROCESS_TRIGGER_ASYNC = "org.jbpm.test.functional.async.TriggerAsyncNodeCompletion";
+    private static final String BPMN_TRIGGER_ASYNC = "org/jbpm/test/functional/async/TriggerAsyncNodeCompletion.bpmn2";
+
+    private ExecutorService executorService;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        executorService = ExecutorServiceFactory.newExecutorService(getEmf());
+        executorService.setInterval(1);
+        executorService.setRetries(0);
```

Comment:
One question here: if we set up the retries to more than one, `handler.onCommandError` is not executed yet (though there are more retries to try) but there's no retryDelay in the context, so probably won't be fired as the scheduled time is the current time, is it fine? Then, process is completed and no execution of the new command is invoked. Perhaps an additional test with these retries scenarios would be valuable, wdyt?